pkg/connector/ticket.go

```
@@ -291,13 +290,20 @@ func (s *ServiceNow) serviceCatalogRequestItemToTicket(ctx context.Context, requ
 		Status: &v2.TicketStatus{
 			Id: requestedItem.State,
 		},
-		Labels:       labels,
 		Url:          s.generateRequestedItemURL(requestedItem),
 		CustomFields: nil,
 		CreatedAt:    timestamppb.New(createdAt),
 		UpdatedAt:    timestamppb.New(updatedAt),
 		CompletedAt:  completedAt,
-	}, nil, nil
+	}
+
+	labels, err := s.client.GetLabelsForRequestedItem(ctx, requestedItem.Id)
+	if err != nil {
+		return t, nil, fmt.Errorf("failed to get labels for requested item %s: %w", requestedItem.Id, err)
+	}
```

What affect does this ordering change have?

---

we may have successfully fetched the ticket, but not the labels, so this still returns the ticket even if we couldn't get the labels

I'm not sure if we really need to return the label error, we don't even use the labels